lib/fluent/plugin/out_forward.rb

```diff
@@ -401,12 +396,15 @@ def on_timer
       }
     end
 
-    def on_heartbeat(sockaddr, msg)
-      if node = @nodes.find {|n| n.sockaddr == sockaddr }
+    def on_udp_heatbeat_response_recv(data, sock)
+      sockaddr = Socket.pack_sockaddr_in(sock.remote_port, sock.remote_host)
+      if node = @nodes.find { |n| n.sockaddr == sockaddr }
         # log.trace "heartbeat arrived", name: node.name, host: node.host, port: node.port
         if node.heartbeat
           @load_balancer.rebuild_weight_array(@nodes)
         end
+      else
+        log.warn("Unknown heartbeat response received from #{soc.remote_host}:#{sock.remote_port}")
```

Comment:
Is `soc` typo?

Reply:
my bad… fixed https://github.com/fluent/fluentd/pull/2503/commits/98417e64f7d8c350c9acadbc775c0d6786122e67